src/Arrays/vector_operations.jl

```diff
@@ -111,7 +111,7 @@ function samedir(u::AbstractVector{N},
             if factor < 0
                 return (false, 0)
             end
-        elseif factor != u[i] / v[i]
+        elseif !isapproxzero(factor - u[i] / v[i])
```

Comment:
```suggestion
        elseif !_isapprox(factor, u[i] / v[i])
```
(the code for `_isapprox` uses both `isapproxzero` and also relative tolerances.. should be safer..)